acceptance/tests/external_facts/non_root_users_default_external_fact_directory.rb

```diff
@@ -0,0 +1,157 @@
+
+  user_base_puppetlabs_dir = get_user_puppetlabs_dir(user_home, agent['platform'])
+  user_puppetlabs_facts_dir = get_user_puppetlabs_facts_dir(user_home, agent['platform'])
+  user_puppetlabs_facts_path = "#{user_puppetlabs_facts_dir}/test.yaml"
+
+  step "Agent #{agent}: figure out facter program location"
+  facter_path = on(agent, "which facter").stdout.chomp
+
+
+  step "Agent #{agent}: create facts directory (#{user_facts_dir})"
+  on(agent, "rm -rf #{user_facts_dir}")
+  on(agent, "mkdir -p #{user_facts_dir}")
+
+  step "Agent #{agent}: create and resolve a custom fact in #{user_facts_dir}"
+  create_remote_file(agent, user_facts_path, ext_user_fact('USER_TEST_FACTER'))
+
+  step "Agent #{agent}: chown and chmod the facts to the user #{non_root_user}"
+  on(agent, "chown -R #{non_root_user} #{user_base_facts_dir}")
+  on(agent, "chmod -R a+rx #{user_base_facts_dir}")
+
+  step "Agent #{agent}: run facter as #{non_root_user} and make sure we get the fact"
+  on(agent, %Q[su #{non_root_user} -c "#{facter_path} test"]) do
+    assert_match(/USER_TEST_FACTER/, stdout, "Fact from #{user_facts_dir} did not resolve correctly")
+  end
+
+  step "Agent #{agent}: remove #{user_facts_path}"
+  on(agent, "rm -rf #{user_facts_path}")
+
+
+  step "Agent #{agent}: create facts directory (#{user_puppetlabs_facts_dir})"
+  on(agent, "rm -rf #{user_puppetlabs_facts_dir}")
+  on(agent, "mkdir -p #{user_puppetlabs_facts_dir}")
+
+  step "Agent #{agent}: create and resolve a custom fact in #{user_puppetlabs_facts_dir}"
+  create_remote_file(agent, user_puppetlabs_facts_path, ext_user_fact('USER_TEST_PUPPETLABS'))
+
+  step "Agent #{agent}: chown and chmod the facts to the user #{non_root_user}"
+  on(agent, "chown -R #{non_root_user} #{user_base_puppetlabs_dir}")
+  on(agent, "chmod -R a+rx #{user_base_puppetlabs_dir}")
+
+  step "Agent #{agent}: run facter as #{non_root_user} and make sure we get the fact"
+  on(agent, %Q[su #{non_root_user} -c "#{facter_path} test"]) do
+    assert_match(/USER_TEST_PUPPETLABS/, stdout, "Fact from #{user_puppetlabs_facts_dir} did not resolve correctly")
+  end
+
+
+  step "Agent #{agent}: create and resolve a custom fact in #{user_puppetlabs_facts_dir}"
+  create_remote_file(agent, user_facts_path, ext_user_fact('USER_PRECEDENCE_FACTER'))
+  create_remote_file(agent, user_puppetlabs_facts_path, ext_user_fact('USER_PRECEDENCE_PUPPETLABS'))
+
+  step "Agent #{agent}: chown and chmod the facts to the user #{non_root_user}"
+  on(agent, "chown -R #{non_root_user} #{user_base_facts_dir} #{user_base_puppetlabs_dir}")
+  on(agent, "chmod -R a+rx #{user_base_facts_dir} #{user_base_puppetlabs_dir}")
+
+  step "Agent #{agent}: run facter as #{non_root_user} and .facter will take precedence over .puppetlabs"
+  on(agent, %Q[su #{non_root_user} -c "#{facter_path} test"]) do
+    assert_match(/USER_PRECEDENCE_FACTER/, stdout, "Fact from #{user_puppetlabs_facts_dir} did not resolve correctly")
+  end
+
+  teardown do
```

Comment:
There appears to be a 60/40 split on beginning or end for putting teardown in the tests in facter. I'll work on converting them over as I go.